Upgrade to GHC 8

emhoracek opened the pull request.

emhoracek committed:
Upgrade to GHC 8

emhoracek committed:
fn-extra works with base > 5

emhoracek committed:
Update version number

emhoracek committed:
Update Stack version for Circle

emhoracek committed:
Build GHC 8 version in Circle

emhoracek committed:
Comment out tests

huggablemonad commented on fn-extra/src/Web/Fn/Extra/Heist.hs:
```diff
@@ -89,11 +93,16 @@ heistInit :: HeistContext ctxt =>
+                                   & hcNamespace .~ ""
+     #if MIN_VERSION_heist(1,0,0)
+     initHeist config
+     #else
```
I'm getting a `parse error on input 'else'`. It seems to go away if the CPP directives aren't indented. I'm building this in a Cabal sandbox (not using `stack`) with GHC 8.0.2.

emhoracek replied:
Bless you, @huggablemonad! It works fine on my machine so I was puzzled why it was failing in CI! Would you be willing to pull the branch and add your changes?

huggablemonad committed:
Remove indentation from CPP directives

This fixes the build failures in CI because of a `parse error on input
'else'`.

emhoracek committed:
Merge pull request #13 from huggablemonad/upgrade-to-ghc8

Remove indentation from CPP directives

emhoracek committed:
Add tests back to circle
